### PR TITLE
feat: search squad members through query

### PR DESCRIPTION
**Pull request by sshanzel**
Fetch members through a query.

- New optional parameter on source members query that if it is present, we will return filtered users based on the value.
- The search query here is based on our search query for user mention.

### CODE REVIEW

File: src/schema/sources.ts [hunk 1]
```diff
@@ -926,13 +933,23 @@ export const resolvers: IResolvers<any, Context> = {
...
-
             .addOrderBy(
               graphorm.mappings.SourceMember.fields.roleRank.select as string,
               'DESC',
             )
             .addOrderBy(`${builder.alias}."createdAt"`, 'DESC');
 
+          if (query) {
+            builder.queryBuilder = builder.queryBuilder
+              .innerJoin(User, 'u', `${builder.alias}."userId" = u.id`)
```
[the remark this answers is not in the document]
Comment (idoshamun) [reply src/schema/sources.ts:944]: It's not possible at the moment but good catch!

File: src/schema/sources.ts [hunk 2]
```diff
@@ -926,13 +933,23 @@ export const resolvers: IResolvers<any, Context> = {
...
+              .andWhere(
+                `(REPLACE(u.name, ' ', '') ILIKE :name OR  u.username ILIKE :name)`,
```
Comment (idoshamun) [inline src/schema/sources.ts:946]: My concern is that it is not indexed properly but that's fine for now I guess.
Anyway, found a double space:

```suggestion
                `(REPLACE(u.name, ' ', '') ILIKE :name OR u.username ILIKE :name)`,
```

File: src/schema/sources.ts [hunk 3]
```diff
@@ -926,13 +933,23 @@ export const resolvers: IResolvers<any, Context> = {
...
+                { name: `${query}%` },
+              );
+          }
+
+          console.log(builder.queryBuilder.getQuery());
```
Comment (idoshamun) [inline src/schema/sources.ts:951]: Can be removed

Comment (sshanzel) [reply src/schema/sources.ts:951]: Oh, right. Debugging 🤦

**Review by idoshamun — APPROVED** [review]

File: src/schema/sources.ts (hunk 2 above)
Comment (sshanzel) [reply src/schema/sources.ts]: Yep, I agree. I think it should be fine for now and the pool of users won't be too large as most members don't exceed 1k, yet.

**Pull request merged**